File: software_design_and_engineering/Enhanced_python/task_test.py
import unittest
from task import Task, IllegalArgumentException


class TestTask(unittest.TestCase):
    test_id = "0123456789"
    test_name = "TaskName<21"
    test_description = "Description<51"

    def test_create_valid_task(self):
        task = Task(self.test_id, self.test_name, self.test_description)
        self.assertEquals(task.get_task_id(), self.test_id)
        self.assertEquals(task.get_task_name(), self.test_name)
        self.assertEquals(task > get_task_description(), self.test_description)

    def test_task_id_too_long(self):
        self.assertRaises(IllegalArgumentException,
                          Task("1234098230498230498230498230498", self.test_name, self.test_description))

    def test_task_name_too_long(self):
        self.assertRaises(IllegalArgumentException,
                          Task(self.test_id,
                               "Thisoisdoifjaosidjfoiasdjfoaisdjfoiasjdfoiasjdofiajsodfm jmfioofdiufoasidjfoasdijfoasidfjosad",
                               self.test_description))

    def test_task_description_too_long(self):
        self.assertRaises(IllegalArgumentException,
                          Task(self.test_id, self.test_name,
                               "aosdifjoasdijfoasidjfoaisdjfoaisdjfoiasdjfoiajsdofijasdofijasodifjaosidfjoasdijfioasjdfoiajsdofiasjdfoiasjdofiajsdofijasdofijasodifjaosdifjosadijfoasidjfoaisdjfoisajdfoasd"))

File: software_design_and_engineering/Original_java/TaskTest.java
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class TaskTest {
    String testID = "0123456789";
    String testName = "TaskName<21";
    String testDescription = "Description<51";

    @Test
    public void createValidTask() {
        Task task = new Task(testID, testName, testDescription);

        assertEquals(task.getTaskId(), testID);
        assertEquals(task.getTaskName(), testName);
        assertEquals(task.getTaskDescription(), testDescription);
        System.out.println(task.toString());
    }

    @Test(expected = IllegalArgumentException.class)
    public void testTaskTooLong() {
        Task task = new Task("1234098230498230498230498230498", testName, testDescription);
    }

    @Test(expected = IllegalArgumentException.class)
    public void testTaskNameTooLong() {
        Task task = new Task(testID, "Thisoisdoifjaosidjfoiasdjfoaisdjfoiasjdfoiasjdofiajsodfm jmfioofdiufoasidjfoasdijfoasidfjosad", testDescription);
    }

    @Test(expected = IllegalArgumentException.class)
    public void testTaskDescriptionTooLong() {
        Task task = new Task(testID, testName, "aosdifjoasdijfoasidjfoaisdjfoaisdjfoiasdjfoiajsdofijasdofijasodifjaosidfjoasdijfioasjdfoiajsdofiasjdfoiasjdofiajsdofijasdofijasodifjaosdifjosadijfoasidjfoaisdjfoisajdfoasd");
    }

}
